source/common/http/http2/codec_impl.cc

```diff
@@ -490,10 +467,7 @@ void ConnectionImpl::StreamImpl::encodeDataHelper(Buffer::Instance& data, bool e
     data_deferred_ = false;
   }
 
-  auto status = parent_.sendPendingFrames();
-  // See comment in the `encodeHeadersBase()` method about this RELEASE_ASSERT.
-  RELEASE_ASSERT(status.ok(), "sendPendingFrames() failure in non dispatching context");
-  parent_.checkProtocolConstraintViolation();
+  parent_.sendPendingFramesAndHandleError();
```

Comment:
question before I review farther, does it make sense to continue processing if an error handler has been scheduled? Most of these are the last function called.

Also, where we sendPendingFramesAndHandleError, are we confident we have tests for error handling at each call site?  Would it make sense to always 
if (sendPendingFramesAndHandleError()) {
  return;
}
so we can coverage-verify the error handling is tested?  Or given these replace release asserts are we asserting they can't happen and mainly hoping they don't cause issues?

Reply:
Yes, I think in some cases additional processing, like scheduling timers is not needed. I have changed these paths.
I have added in last few PRs 16 new integration test cases for all invocations of the `sendPendingFrames()`. To the best of my analysis all call sites in all possible contexts are covered. It does not mean that I'm right of course. So I've added the `if` clause as you have suggested and will check the coverage output.